SkietbaanBE/SkietbaanBE/Controllers/CompetitionController.cs

```diff
@@ -58,14 +60,20 @@ public async Task<Competition> CompetitionGetByName(string Name)
         [HttpPost]
         public async Task<IActionResult> AddCompetition([FromBody]Competition comp)
         {
-            if (!ModelState.IsValid)
+            if (ModelState.IsValid)
             {
-                return BadRequest(ModelState);
+                Competition dbComp = _context.Competitions.FirstOrDefault(C => C.Name == comp.Name);
+                if (dbComp != null)
+                    return new BadRequestObjectResult(comp.Name+"already exists");
+
+                _notificationMessages.CompetitionNotification(_context, comp);
+                await _context.AddAsync(comp);
+                await _context.SaveChangesAsync();
+                return Ok("Competition Added!!!!!!!");
```

Comment:
How do you know for sure that the 2 previous awaits were completed successfully? Perhaps check the success of them first before confirming to the user that it was successful

Reply:
Hi Rashad, most backend devs didn't mind for not checking the outcome, and I do understand what you do mean. I am new on the backend, and my LINQ is not as good yet. I will google, and learn how to check the bool for async functions.

Reply:
You can just assign a local variable to the result of the async task and check the result and do the handling in this class